File: Python/0801-0850/814. Binary Tree Pruning.py
# https://leetcode.com/problems/binary-tree-pruning/

class Solution:
    def pruneTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root == None:
            return None

        root.left = self.pruneTree(root.left)
        root.right = self.pruneTree(root.right)

        if root.val == 0 and root.left == None and root.right == None:
            return None

        return root

File: Java/0801-0850/814. Binary Tree Pruning.java
// https://leetcode.com/problems/binary-tree-pruning/

class Solution {
    public TreeNode pruneTree(TreeNode root) {
        if(root == null) return null;

        root.left = pruneTree(root.left);
        root.right = pruneTree(root.right);

        if(root.val == 0)
            if (root.left == null && root.right == null) return null;

        return root;
    }
}
